### DISCUSSION TITLE
How to read a chest inventory

### DISCUSSION THREAD

**Discussion by CesarCruzUtec** [cc-tweaked/CC-Tweaked#1833] [Q&A] [answered]
I recently discovered ComputerCraft for 1.20.1 and decided to integrate it into my Sky Greg modpack. Prior to this, I began learning the basics through YouTube videos and the wiki. The first functionality I tested was reading a chest inventory, for which I copied the following code:
![image](https://github.com/cc-tweaked/CC-Tweaked/assets/67798431/97d7e802-6aa6-4938-9880-1de2b8e82f77)
I attempted to run the code in the modpack, but it failed and produced "nil" errors. Initially, I suspected a mod conflict, so I set up a new instance with only ComputerCraft installed, yet the issue persisted.

In my modpack, I will have numerous chests scattered around, and I've learned that networking cables and wired modems are necessary. Therefore, I am testing with the following setup.
![image](https://github.com/cc-tweaked/CC-Tweaked/assets/67798431/72bd3433-ba76-4bd4-9b62-6b652d2cee16)

I've been running some code for testing, but unfortunately, it keeps failing. I've tried many methods to get the code to work and have consulted the official Lua API reference, but I still can't seem to make it function correctly. I'm not sure if I'm doing something wrong. I've come here seeking your help, and I'm willing to provide any additional information if needed.
![image](https://github.com/cc-tweaked/CC-Tweaked/assets/67798431/77f5148f-b422-4f70-977a-95615c775e5f)

**Comment by piprett** [comment, marked as answer]
As your lua repl says, "To access local variables in later inputs, remove the local keyword.". The reason they use `local` in the documentation, is because the code is put into a file and ran. You are running the code is the repl, which works a bit differently.

Additionally, in the example you can see they use `peripheral.find("minecraft:chest")`, whereas you use `peripheral.find("minecraft:chest_0")`. If you wish to wrap the specific peripheral, use `peripheral.wrap("minecraft:chest_0")`, or if you want to automatically find it use `peripheral.find("minecraft:chest")`.

**Reply by CesarCruzUtec** [reply]
Thank you for the answer. As you mentioned, my error was using the 'find' method instead of 'wrap'. I spent two hours reading and performing trial and error until I managed to make it work.
